src/Module.ts

```diff
@@ -0,0 +1,82 @@
+import { Modal } from "./component/layout/Modal";
+import { Filter } from "./domain/Filter";
+import {
+  convertStorageToLocal,
+  readFoodList,
+  showConvertedItem,
+} from "./domain/handler/FoodItemHandler";
+import { isFavoriteState } from "./domain/handler/TabButtonHandler";
+import { FoodForm } from "./pages/FoodForm";
+import { FilterType } from "./types/domain/FilterType";
+
+export class Module {
+  #filter;
+
+  constructor(filter: Filter) {
+    this.#filter = filter;
+  }
+
+  init() {
+    this.#setFoodFormMoal();
+    this.#setFilteredItems();
+    this.#setFavoriteButton();
+  }
+
+  // FoodForm 생성
+  #setFoodFormMoal() {
+    document.querySelector(".gnb__button")?.addEventListener("click", () => {
+      const formContainer = document.createElement("div");
+      formContainer.innerHTML = `<h2 class="modal-title text-title">새로운 음식점</h2>`;
+      formContainer.appendChild(FoodForm({ filter: this.#filter }));
+      Modal.setContent({ filter: this.#filter, modalContent: formContainer });
+      Modal.open();
+    });
+  }
+
+  // Filter 기능 설정
+  #setFilteredItems() {
+    ["category", "sorting"].forEach((name) => {
+      document
+        .querySelector(`select[name=${name}]`)
+        ?.addEventListener("change", () => {
+          const previousFoodList = readFoodList({
+            favoriteFilter: isFavoriteState(),
+          });
+
+          const filteredItems = this.#filter.chageFilter({
+            foodList: previousFoodList,
+            filter: name as FilterType,
+          });
```

Comment:
이렇게 바꾸면, 타입 단언 문제가 해결되겠어요. 
왜 그런지 tuple vs array 를 학습해보아요~!


```suggestion
    const filterTabs = ["category", "sorting"] as const
    filterTabs.forEach((name) => {
      document
        .querySelector(`select[name=${name}]`)
        ?.addEventListener("change", () => {
          const previousFoodList = readFoodList({
            favoriteFilter: isFavoriteState(),
          });

          const filteredItems = this.#filter.chageFilter({
            foodList: previousFoodList,
            filter: name,
          });
```